feat(uptime): add open beta documentation for uptime (#11420)
getsentry/sentry-docs: docs/product/alerts/uptime-monitoring/automatic-detection.mdx

description: "Learn how automatic detection of uptime monitoring works."
---

<Include name="feature-stage-beta-uptime.mdx" />

> The automatic detection of uptime alerts sets up an uptime alert for the most frequently encountered hostname in all URLs of your error data. This helps ensure that your most critical hostname is continuously monitored, enhancing the reliability and availability of your web services.

Reviewer: ```suggestion
In order to be able to automatically detect uptime alerts, we analyze all the URLs detected in your project's captured error data to find the hostname that appears most frequently. This helps ensure that your most critical hostname is continuously monitored, enhancing the reliability and availability of your web services. We then create an uptime alert if it passes our [uptime check criteria](/product/alerts/uptime-monitoring/#uptime-check-criteria). Automatic uptime checks are configured to run once a minute as GET requests.
```

> ## How It Works

Reviewer: ```suggestion

```

> We analyze all the URLs detected in your project's captured error data to find the hostname that appears most frequently. We then create an uptime alert if it passes our [uptime check criteria](/product/alerts/uptime-monitoring/#uptime-check-criteria). Automatic uptime checks are configured to run once a minute as GET requests.

Reviewer: ```suggestion

```

> To avoid creating flaky alerts, the hostname undergoes an "onboarding period" of three days. During this period, we send HTTP requests to the hostname every hour. If the request fails at least three times, the hostname is dropped and re-evaluated after seven days.

Reviewer: ```suggestion
To avoid creating flaky alerts, the hostname undergoes an "onboarding period" of three days. During this period, we send HTTP requests to the hostname every hour. If the request fails three times or more, the hostname will be dropped and re-evaluated after seven days.
```

<Alert level="info">

>   Sentry will execute uptime checks against the hostname root path of the most
>   frequently seen URLs. For example, if the most seen URL in your events is `GET
>   https://www.example.com/docs/introduction` the check will be `GET
>   https://www.example.com/`.

Reviewer: ```suggestion
 Sentry will execute uptime checks against the hostname root path of the most frequently-seen URLs. For example, if the most frequently-seen URL in your events is `GET https://www.example.com/docs/introduction`, the check will be `GET https://www.example.com/`.
```
I'm not sure we need this alert since we already have this information further up at the top... might consider deleting it.

</Alert>

> After an automatically detected uptime alert is created, you can [customize its configuration](/product/alerts/create-alerts/uptime-alert-config/) such as the HTTP request method, headers, and request body.

Reviewer: ```suggestion
Once an automatically detected uptime alert has been created, you'll be able to [customize its configuration](/product/alerts/create-alerts/uptime-alert-config/), including the HTTP request method, headers, and request body.
```

## Disabling Automatic Detection

> If you prefer not to use automatically detected uptime alerts, you have two options to disable them:

Reviewer: ```suggestion
If you prefer not to use automatically detected uptime alerts, you have two options for disabling them:
```

```

<Alert level="warning">

>   **Note:** The `robots.txt` file will only prevent Sentry from detecting URLs
>   that it encounters in the future. Existing URLs that have already been
>   detected will continue to have uptime alerts unless they are manually deleted.

Reviewer: ```suggestion
**Note:** The `robots.txt` file only prevents Sentry from detecting URLs that it encounters after its been implemented. Existing URLs that have already been detected will continue to have uptime alerts unless they're manually deleted.
```
